File: mobile/lib/core/util/money_format.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter_bloc/flutter_bloc.dart';
import 'package:intl/intl.dart';
import 'package:twoaxis_finance/features/user/presentation/bloc/user_cubit.dart';
import 'package:twoaxis_finance/features/user/presentation/bloc/user_state.dart';

String formatMoneyWithContext(BuildContext context, num value) {
  var state = context.read<UserCubit>().state;
  String currency = 'USD';
  if (state is UserStateAuthenticated) {
    currency = state.user.currency;
  }
  var format = NumberFormat.simpleCurrency(name: currency);
  return format.format(value);
}

File: web/src/utils/moneyFormat.ts
export function formatMoney(value: number, currency: string = 'USD'): string {
  const num = typeof value === 'number' && !isNaN(value) ? value : (parseFloat(String(value)) || 0);
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: currency || 'USD',
  }).format(num);
}
